### DISCUSSION TITLE
Supabase with Svelte

### DISCUSSION THREAD

**Discussion by natileeSquires-lab** [supabase/supabase#5860] [Questions] [answered]
I am trying to code a select statement in my website with Svelte and the select is not recognizing my argument - what is the syntax to get supabase to recognize the argument. The function is not working.

<script>
    import supabase from '$lib/db'

  let selected;

    // function to get data from supabase
    async function getDataLevel() {
        const { data, error} = await supabase.from("tablename").select()
        .eq('class',{selected})
            if (error) throw new Error(error.message)
          return data
      }

      // function for dropdown
    function selectionChanged(event){
        if(!event.target.value) return;
    }

</script>

  {#await getDataLevel()}
    <p>Fetching data...</p>
  {:then data}
  <button>
    <select bind:value = {selected} on:change={selectionChanged}>
    {#each data as class}
        <option value={class.index_value}>{class.spell_title}  </option>
        {/each}
            </select></button>
  {:catch error}
    <p>There is an error while fetching the data:</p>
    <pre>{error}</pre>
  {/await}
{/if}

**Comment by natileeSquires-lab** [comment]
Cleaned up the code as above was hard to read.

<script> 
import supabase from '$lib/db' 
let selected; 
// function to get data from supabase 
async function getDataLevel() { const { data, error} = await supabase.from("tablename")
.select() .
eq('class',{selected}) 
if (error) throw new Error(error.message)

 return data } 
// function for dropdown 
function selectionChanged(event){ if(!event.target.value) return; } 

</script>

{#await getDataLevel()}
Fetching data...

{:then data}

{#each data as class} {class.spell_title} {/each}
{:catch error}
There is an error while fetching the data:
{error}
{/await}
{/if}

**Comment by GaryAustin1** [comment, marked as answer]
Your code is still hard to read.  Is selected being set to anything?  Looks like it is just let selected; ...
Edit:  Also why "{selected}" versus just "selected" in the eq filter?

**Reply by natileeSquires-lab** [reply]
I will try the basic code again - every time I type the code, the formatting changes: 


    // function to get class from supabase
    async function getDataLevel() {
        const { data, error} = await supabase.from("Class").select('*')
        .lt('class_level', selected)
            if (error) throw new Error(error.message)
          return data
      }

This is my function and the variable(parameter) called selected does not work - I get the whole table not just a selected class level.

**Reply by GaryAustin1** [reply]
You are now showing a .lt filter which will return all classes less than selected.

**Reply by GaryAustin1** [reply]
Can you put a console.log ('selected =',selected) right before the supabase call?  Also just hardcode a number where you have selected and see if that works (it should).

**Reply by natileeSquires-lab** [reply]
If I hard code it works just fine.  When I do a console log the selected variable does show the correct data.  It just seems that the function select statement does not read the variable.  I did research with supabase but did not find any syntax example when using a variable to select data from supabase.

**Reply by GaryAustin1** [reply]
```
    let myid = 10
    const { data, error: } = await supabase
        .from('messages')
        .select('*')
        .lt('id',myid)
```
works

**Reply by natileeSquires-lab** [reply]
Thanks for your help.

**Comment by natileeSquires-lab** [comment]
Yes, that is what I want but the selected variable does not work.  I get all classes whether less than, equal to or greater than. The function is not interpreting the variable.
